main_test.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/ONSdigital/dp-permissions-api/features/steps"
+	"os"
+	"testing"
+
+	componenttest "github.com/ONSdigital/dp-component-test"
+	"github.com/cucumber/godog"
+	"github.com/cucumber/godog/colors"
+)
+
+// Mongo version here is overridden in the pipeline by the URL provided in the component.sh
+const MongoVersion = "4.0.23"
+const DatabaseName = "testing"
+
+var componentFlag = flag.Bool("component", false, "perform component tests")
```

Comment:
I believe you need to call `flag.Parse()` in test main in order to retrieve the value specified by the user. As it is I believe this will always return the default value set here. 

https://pkg.go.dev/flag#Parse